default_handler.go

```diff
@@ -126,18 +126,32 @@ func (m exportedMethod) Call(args ...interface{}) ([]interface{}, error) {
 	}
 
 	ret := m.Value.Call(params)
-
-	err := ret[t.NumOut()-1].Interface().(*Error)
-	ret = ret[:t.NumOut()-1]
+	var err error
+	nilErr := false // The reflection will find almost-nils, let's only pass back clean ones!
+	if t.NumOut() > 0 {
+		if e, ok := ret[t.NumOut()-1].Interface().(*Error); ok { // godbus *Error
+			nilErr = ret[t.NumOut()-1].IsNil()
+			ret = ret[:t.NumOut()-1]
+			err = e
+		} else if ret[t.NumOut()-1].Type().Implements(reflect.TypeOf((*error)(nil)).Elem()) { // Go error
```

Comment:
You might consider creating an "errorType" definition here: https://github.com/godbus/dbus/blob/1a8824955e97cb5b648855b2ba00cbf7a0913867/dbus.go#L10 so this is only allocated once.